internal/pb/pb.go

```diff
@@ -46,9 +47,24 @@ type progressBar struct {
 }
 
 // NewProgressBar creates a new progress bar.
-func NewProgressBar() *ProgressBar {
+func NewProgressBar(writers ...io.Writer) *ProgressBar {
+	opts := []mpbv8.ContainerOption{
+		mpbv8.WithAutoRefresh(),
+		mpbv8.WithWidth(60),
+		mpbv8.WithRefreshRate(300 * time.Millisecond),
+	}
+
+	// If no writer specified, use stdout.
+	if len(writers) == 0 {
+		opts = append(opts, mpbv8.WithOutput(os.Stdout))
+	} else if len(writers) == 1 {
+		opts = append(opts, mpbv8.WithOutput(writers[0]))
```

Comment:
Can we also use io.MultiWriter here?

Reply:
We set the default `ProgressWriter` as `os.Stdout` in the pull config, and pass it from progress bar `New`, if here we're using the `io.MultiWriter` to wrap the `os.Stdout`, then it will be not recognized as the terminal output which will affect the render of the progress bar. BTW, usually only needs one writer, here using a mutable type is only to allow the option of not passing a value when creating a new instance. Normally, there will only be one value.

Reply:
SGTM, thanks!